rfd/0215-mwi-terraform-provider.md

```diff
@@ -0,0 +1,357 @@
+which currently use older versions of Terraform.
+
+Therefore, we should support both data sources and ephemeral resources at least
+until support is more common-place. This will enable customers with more recent
+versions of Terraform to use the more secure option. In 6-12 months, we can 
+revisit this decision and consider deprecating support for data sources.
+
+### Which provider to use
+
+One key decision is where to build this functionality. Largely, we have three
+choices:
+
+- Build this functionality within the existing Teleport Terraform provider.
+  - Positives:
+    - No additional build/release pipelines to maintain.
+    - Customers leveraging this functionality and managing a Teleport cluster 
+      within the same plan/apply only need to configure a single provider.
+  - Negatives:
+    - Existing Teleport terraform provider has a significant degree of tech
+      debt, and due to the different nature of this functionality, we're likely
+      to run into unforeseen edge-cases within the core implementation of the 
+      Terraform provider.
+    - Existing Terraform provider consists largely of generated code, whereas
+      this functionality will be largely hand-written. Mixing generated and 
+      hand-written code can make enforcing standards more difficult.
+    - Existing Terraform provider uses a very old version of the Terraform SDK
+      which makes compatability with other tools like Pulumi more challenging.
+    - Existing Terraform provider has a large number of configuration
+      parameters which would not be compatible with this functionality, this
+      poses a risk of creating a poor configuration UX.
+- Build all variants of this functionality into a new Terraform provider
+  - Positives:
+    - No inherited tech debt from existing provider & can leverage latest
+      versions of the Terraform SDK.
+    - Separation of hand-written and generated code.
+  - Negatives:
+    - Additional build and release infrastructure to maintain.
+    - Additional codebase to maintain.
+- Build each variant of this functionality as its own Terraform provider
+  - Positives:
+    - No inherited tech debt from existing provider & can leverage latest
+      versions of the Terraform SDK.
+    - Separation of hand-written and generated code.
+    - Separation of variants would allow for smaller binary sizes.
+  - Negatives:
+    - Much more build and release infrastructure to maintain. This would need 
+      to be handled each time we added a new variant.
+
+Overall, the most appropriate approach seems to be introducing a single new
+Terraform provider for this functionality.
+
+## UX
+
+Before using the data sources or ephemeral resources, the user must configure
+the provider itself. To do so, they will provide the details required to
+connect and authenticate to the Teleport cluster.
+
+Configuring the MWI provider:
+
+```hcl
+provider "mwi" {
+  proxy_addr  = "example.teleport.sh:443"
+  join_method = "terraform_cloud"
```

Comment:
It's tempting to pull these join fields into an "onboarding" object to match the tbot yaml config. WDYT?